src/sentry/sentry_apps/api/bases/sentryapps.py

```diff
@@ -0,0 +1,515 @@
+        return super().has_permission(request, *args, **kwargs)
+
+    def has_object_permission(self, request: Request, view, installation):
+        if not hasattr(request, "user") or not request.user or not request.user.is_authenticated:
+            return False
+
+        self.determine_access(request, installation.organization_id)
+
+        if superuser_has_permission(request):
+            return True
+
+        # if user is an app, make sure it's for that same app
+        if request.user.is_sentry_app:
+            return request.user.id == installation.sentry_app.proxy_user_id
+
+        org_context = organization_service.get_organization_by_id(
+            id=installation.organization_id,
+            user_id=request.user.id,
+            include_teams=False,
+            include_projects=False,
+        )
+        if (
+            not org_context
+            or not org_context.member
+            or org_context.organization.status != OrganizationStatus.ACTIVE
+        ):
+            raise Http404
+
+        assert request.method, "method must be present in request to get permissions"
+        return ensure_scoped_permission(request, self.scope_map.get(request.method))
+
+
+class SentryAppInstallationBaseEndpoint(IntegrationPlatformEndpoint):
+    permission_classes: tuple[type[BasePermission], ...] = (SentryAppInstallationPermission,)
+
+    def convert_args(self, request: Request, uuid, *args, **kwargs):
+        installations = app_service.get_many(filter=dict(uuids=[uuid]))
+        installation = installations[0] if installations else None
+        if installation is None:
+            raise Http404
+
+        self.check_object_permissions(request, installation)
+
+        Scope.get_isolation_scope().set_tag("sentry_app_installation", installation.uuid)
+
+        kwargs["installation"] = installation
+        return (args, kwargs)
+
+
+class SentryAppInstallationExternalIssuePermission(SentryAppInstallationPermission):
+    scope_map = {
+        "POST": ("event:read", "event:write", "event:admin"),
+        "DELETE": ("event:admin",),
+    }
+
+
+class SentryAppInstallationExternalIssueBaseEndpoint(SentryAppInstallationBaseEndpoint):
+    permission_classes = (SentryAppInstallationExternalIssuePermission,)
+
+
+class SentryAppAuthorizationsPermission(SentryPermission):
+    def has_object_permission(self, request: Request, view, installation):
+        if not hasattr(request, "user") or not request.user:
+            return False
+
+        installation_org_context = organization_service.get_organization_by_id(
+            id=installation.organization_id, user_id=request.user.id
+        )
+        assert installation_org_context, "organization for installation was not found"
+        self.determine_access(request, installation_org_context)
+
+        assert (
+            request.user.is_authenticated
+        ), "user must be authenticated to check if they're a sentry app"
+        if not request.user.is_sentry_app:
```

Comment:
This could result in additional errors being captured if unauthenticated traffic is getting to sentryapps endpoints (which it likely is).

```suggestion
        if not request.user.is_authenticated or not request.user.is_sentry_app:
```

Would something like this work?